src/ess/powder/correction.py

```diff
@@ -257,6 +257,86 @@ def _shallow_copy(da: sc.DataArray) -> sc.DataArray:
     return out
 
 
+def compute_pdf_from_structure_factor(
+    s: sc.DataArray,
+    *,
+    rho: sc.Variable,
+    r: sc.Variable,
+    self_scattering: sc.Variable,
+    use_filter: bool,
+) -> sc.DataArray:
+    '''
+    Computes the pair-distribution-function :math:`D(r)` as defined in
+    `Review: Pair distribution functions from neutron total scattering for the study of local structure in disordered materials <https://www.sciencedirect.com/science/article/pii/S2773183922000374>`_
+    from the overall scattering function :math:`S(Q)`.
+
+    The inputs to the algorithm are:
+
+    * A histogram representing :math:`S(Q)` with :math:`N` bins on a bin-edge grid with :math:`N+1` edges :math:`Q_j` for :math:`j=0\\ldots N`.
+    * The density :math:`\\rho`, the total number of atoms per volume.
+    * The structure independent coherent scattering amplitude :math:`S_{self} = \\sum_m c_m b_m^2` where the sum is over the atomic species in the sample, :math:`c_m` is the fraction of atoms that belong to species :math:`m` and :math:`b_m` is the size of scattering from an atom of that species.
+    * The bin-edge grid over :math:`r` the output histogram representing :math:`D(r)` will be computed on.
```

Comment:
Yes it felt a bit like that to me to. But it also seemed reasonable to have the definitions before the actual description of the procedure. Otherwise the user would have to go to the parameter section to understand what the symbols in the expression for $D$ means. It seemed easier to just have some duplication here.